src/plugins/data_views/public/services/has_data.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart, HttpStart } from 'kibana/public';
+import { FLEET_ASSETS_TO_IGNORE } from '../../common';
+import { HasDataViewsResponse, IndicesResponse, IndicesResponseModified } from '../';
+
+export class HasData {
+  private removeAliases = (source: IndicesResponseModified): boolean => !source.item.indices;
+
+  private isUserDataIndex = (source: IndicesResponseModified): boolean => {
+    // filter out indices that start with `.`
+    if (source.name.startsWith('.')) return false;
+
+    // filter out sources from FLEET_ASSETS_TO_IGNORE
+    for (const key in FLEET_ASSETS_TO_IGNORE) {
+      if (source.name === (FLEET_ASSETS_TO_IGNORE as any)[key]) return false;
+    }
+
+    // filter out empty sources created by apm server
+    if (source.name.startsWith('apm-')) return false;
```

Comment:
Can't this be a part of `FLEET_ASSETS_TO_IGNORE`?

Reply:
There is a chance that this actually not needed since 8.0, need to double-check
More context: https://github.com/elastic/kibana/issues/110367#issuecomment-1061873106

Reply:
Create an issue to check everything and clean up https://github.com/elastic/kibana/issues/127869